File: JavaScript/sumOfRange/sumOfRange.js
function range(arg1, arg2, step = 1) {
    let range = [];

    for (let iterator = arg1; iterator < arg2 + 1; iterator += step) {
        range.push(iterator);
    }

    return range;
}

// caps below a range 0, 10_000 call

function recursiveSum(numArr, sumNum = 0) {
    numArr.length < 1 ? console.log(sumNum) : recursiveSum(numArr.slice(1), sumNum + numArr[0]);

}

// easily handles 100_000_000 iterations

function iterativeSum(numArr) {
    let sumNum = 0;

    numArr.forEach( (num) => {
        sumNum += num;
    });

    console.log(sumNum);
}



iterativeSum(range(0, 99));
recursiveSum(range(0, 7000));

File: Python/sum_of_range/sum_of_range.py
def manual_range(min, max, step = 1):
    collected_range = []
    iterator = min

    while iterator <= max:
        collected_range.append(iterator)
        iterator += step

    return collected_range

def recursive_sum(num_arr, num_sum = 0):
    if len(num_arr) < 1: return print(num_sum)

    recursive_sum(num_arr[1:], num_sum + num_arr[0])


def iterative_sum(num_arr):
    sum_num = 0

    for num in num_arr:
        sum_num += num

    print(sum_num)

recursive_sum(manual_range(0, 100))
iterative_sum(manual_range(0, 100))

# Compare performance. Which can handle a wider range? Why? Which should we use in this case? Consider readability versus functionality.
